File: Exercise/Medium/06. Find the Missing Number in a Sequence/Code/Find the Missing Number in a Sequence.js
function findMissingNumber(arr) {
    const expectedSum = (arr[0] + arr[arr.length - 1]) * (arr.length + 1) / 2;
    const actualSum = arr.reduce((sum, num) => sum + num, 0);
    return expectedSum - actualSum;
}

console.log(findMissingNumber([1, 2, 4, 5]));

File: Exercise/Medium/06. Find the Missing Number in a Sequence/Code/Find the Missing Number in a Sequence.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program {
    static int FindMissingNumber(List<int> arr) {
        int expectedSum = (arr.First() + arr.Last()) * (arr.Count + 1) / 2;
        int actualSum = arr.Sum();
        return expectedSum - actualSum;
    }

    static void Main() {
        string line = Console.ReadLine();
        
        List<int> arr = line.Split(',').Select(int.Parse).ToList();
        
        Console.WriteLine(FindMissingNumber(arr));
    }
}


